opensearch/src/test/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactoryTest.java

```diff
@@ -390,22 +401,239 @@ public void constructArray() {
             }))),
         tupleValue("{\"arrayV\":[{\"info\":\"zz\",\"author\":\"au\"}]}").get("arrayV"));
     assertEquals(
-        new ExprCollectionValue(ImmutableList.of(new ExprTupleValue(
+        new ExprCollectionValue(List.of(new ExprTupleValue(
             new LinkedHashMap<String, ExprValue>() {
               {
                 put("info", stringValue("zz"));
                 put("author", stringValue("au"));
               }
             }))),
-        constructFromObject("arrayV", ImmutableList.of(
+        constructFromObject("arrayV", List.of(
             ImmutableMap.of("info", "zz", "author", "au"))));
   }
 
   @Test
   public void constructArrayOfStrings() {
     assertEquals(new ExprCollectionValue(
-            ImmutableList.of(new ExprStringValue("zz"), new ExprStringValue("au"))),
-        constructFromObject("arrayV", ImmutableList.of("zz", "au")));
+            List.of(stringValue("zz"), stringValue("au"))),
+        constructFromObject("arrayV", List.of("zz", "au")));
+  }
+
+  @Test
+  public void constructNestedArraysOfStrings() {
+    assertEquals(
+        new ExprCollectionValue(
+            List.of(
+                collectionValue(
+                  List.of("zz", "au")
+                ),
+                collectionValue(
+                    List.of("ss")
+                )
+            )
+        ),
+        tupleValueWithArraySupport(
+            "{\"stringV\":["
+                + "[\"zz\", \"au\"],"
+                + "[\"ss\"]"
+                + "]}"
+        ).get("stringV"));
+  }
+
+  @Test
+  public void constructNestedArraysOfStringsReturnsFirstIndex() {
+    assertEquals(
+        stringValue("zz"),
+        tupleValue(
+            "{\"stringV\":["
```

Comment:
what if the data is?
```
{
  "stringV": ["z", ["s"], ["zz", "au"]]
}
```

Reply:
With this array structure the parsed value will be "z". This is the same as prior to this PR. I will add a test with this case.

Reply:
@penghuo your case mostly related to #1300